source/cli/ArgumentParser.ooc

```diff
@@ -0,0 +1,112 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use ooc-collections
+use ooc-base
+import Event
+
+Argument: class {
+	_longIdentifier: Text
+	_shortIdentifier: Char
+	_parameters: Int
+	_action1: Event
+	_action2: Event1<Text>
+	_action3: Event1<VectorList<Text>>
```

Comment:
We do in `Event`, obviously, but I think we can do better here, since the number doesn't describe the number of parameters, but a degree of complexity. So, how about `action`, `textAction`, `listAction`?